core/src/main/java/hudson/FilePath.java

```
@@ -2655,13 +2655,16 @@ private void readFromTar(String name, File baseDir, InputStream in) throws IOExc
                         if (mode != 0 && !Functions.isWindows()) // be defensive
                             _chmod(f, mode);
                     }
+                    files.add(f);
                 }
             }
         } catch (IOException e) {
             throw new IOException("Failed to extract " + name, e);
         } catch (InterruptedException e) {
             Thread.currentThread().interrupt(); // process this later
             throw new IOException("Failed to extract " + name, e);
+        }finally {
+            return files;
```

all exception will be ignored. I would not do that.